### PR TITLE
158037408 prototype the common interface

### PR DESCRIPTION
**Pull request by newlandp**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,16 +16,19 @@
...
     "@angular/compiler": "6.0.3",
-    "@angular/animations": "6.0.3",
+    "@angular/core": "6.0.3",
     "@angular/forms": "6.0.3",
     "@angular/http": "6.0.3",
+    "@angular/material": "^6.2.1",
     "@angular/platform-browser": "6.0.3",
     "@angular/platform-browser-dynamic": "6.0.3",
     "@angular/router": "6.0.3",
+    "@ng-bootstrap/ng-bootstrap": "^2.1.0",
```
Comment (escalonn) [inline package.json]: remove ng-bootstrap dep

File: projects/revaturecloud/forecast/src/lib/bar-chart/bar-chart.component.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  </div>
+  <div class="card-body">
+    <canvas baseChart
+            [datasets]="this.barChartData"
+            [labels]="this.barChartLabels"
+            [options]="this.barChartOptions"
+            [legend]="this.barChartLegend"
+            [chartType]="this.barChartType"
+            (chartHover)="this.chartHovered($event)"
+            (chartClick)="this.chartClicked($event)">
```
Comment (escalonn) [inline projects/revaturecloud/forecast/src/lib/bar-chart/bar-chart.component.html]: should not be necessary to have `this.` in template here or other templates.

File: projects/revaturecloud/forecast/src/lib/forecast.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,57 @@
...
+import {MatTabsModule} from '@angular/material/tabs';
+import {MatCardModule} from '@angular/material/card';
+import {MatIconModule} from '@angular/material/icon';
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import { DummyComponent } from './dummy/dummy.component';
+
+import { By } from '@angular/platform-browser';
+import { RadarChartComponent } from './radar-chart/radar-chart.component';
+import {ChartsModule} from 'ng2-charts';
+import { ForecastModule } from './forecast.module';
```
Comment (escalonn) [inline projects/revaturecloud/forecast/src/lib/forecast.component.spec.ts]: formatting in this file

File: projects/revaturecloud/forecast/src/lib/bar-chart/bar-chart.component.html (hunk 2 above)
Comment (newlandp) [reply projects/revaturecloud/forecast/src/lib/bar-chart/bar-chart.component.html]: have it to be explicit and also the angular code highlighting was giving errors if we didn't have it

**Pull request merged**